homeassistant/components/media_player/ziggo_mediabox_xl.py

```diff
@@ -0,0 +1,175 @@
+    else:
+        _LOGGER.info("Ignoring duplicate Ziggo Mediabox XL %s", host)
+    add_devices(hosts, True)
+
+
+class ZiggoMediaboxXLDevice(MediaPlayerDevice):
+    """Representation of a Ziggo Mediabox XL Device."""
+
+    def __init__(self, mediabox, host, name):
+        """Initialize the device."""
+        # Generate a configuration for the Samsung library
+        self._mediabox = mediabox
+        self._host = host
+        self._name = name
+        self._state = None
+
+    def update(self):
+        """Retrieve the state of the device."""
+        try:
+            if self._mediabox.turned_on():
+                if self._state != STATE_PAUSED:
+                    self._state = STATE_PLAYING
+            else:
+                self._state = STATE_OFF
+        except socket.error:
+            _LOGGER.error("Couldn't fetch state from %s", self._host)
+
+    def send_keys(self, keys):
+        """Send keys to the device and handle exceptions."""
+        try:
+            self._mediabox.send_keys(keys)
+        except socket.error:
+            _LOGGER.error("Couldn't send keys to %s", self._host)
+
+    @property
+    def name(self):
+        """Return the name of the device."""
+        return self._name
+
+    @property
+    def state(self):
+        """Return the state of the device."""
+        return self._state
+
+    @property
+    def source_list(self):
+        """List of available sources (channels)."""
+        return [self._mediabox.channels()[c]
+                for c in sorted(self._mediabox.channels().keys())]
+
+    @property
+    def supported_features(self):
+        """Flag media player features that are supported."""
+        return SUPPORT_ZIGGO
+
+    def turn_on(self):
+        """Turn the media player on."""
+        self.send_keys(['POWER'])
+        self._state = STATE_OFF
+
+    def turn_off(self):
+        """Turn off media player."""
+        self.send_keys(['POWER'])
+        self._state = STATE_ON
+
+    def media_play(self):
+        """Send play command."""
+        self.send_keys(['PLAY'])
+        self._state = STATE_PLAYING
+
+    def media_pause(self):
+        """Send pause command."""
+        self.send_keys(['PAUSE'])
+        self._state = STATE_PAUSED
+
+    def media_play_pause(self):
+        """Simulate play pause media player."""
+        self.send_keys(['PAUSE'])
+        if self._state == STATE_PAUSED:
+            self._state = STATE_PLAYING
+        else:
+            self._state = STATE_PAUSED
+
+    def media_next_track(self):
+        """Channel up."""
+        self.send_keys(['CHAN_UP'])
+        self._state = STATE_PLAYING
+
+    def media_previous_track(self):
+        """Channel down."""
+        self.send_keys(['CHAN_DOWN'])
+        self._state = STATE_PLAYING
+
+    def select_source(self, source):
+        """Select the channel."""
+        if str(source).isdigit():
+            digits = str(source)
+        elif source in self._mediabox.channels().values():
```

Comment:
```py
else:
    digits = next((
        key for key, value in self._mediabox.channels().items()
        if value == source), None)
if digits is None:
    return
self.send_keys(...
```

Reply:
Ah, a more pythonic way to achieve the same. Didn't know about this yet. Thanks :)